blockchain/types/tx_internal_data_smart_contract_execution.go

```diff
@@ -238,7 +239,7 @@ func (t *TxInternalDataSmartContractExecution) String() string {
 func (t *TxInternalDataSmartContractExecution) IntrinsicGas(currentBlockNumber uint64) (uint64, error) {
 	gas := params.TxGasContractExecution
 
-	gasPayloadWithGas, err := IntrinsicGasPayload(gas, t.Payload)
+	gasPayloadWithGas, err := IntrinsicGasPayload(gas, t.Payload, t.GetRecipient() == nil, *fork.Rules(big.NewInt(int64(currentBlockNumber))))
```

Comment:
Why did you leave `t.GetRecipient() == nil` only for contract execution types? It t.GetRecipient() == nil, the tx will fail to be validated.

Reply:
It's my fault. I thought deployment could possible with execution tx types. Fixed it.